curriculum/src/build-curriculum.ts

```diff
@@ -335,9 +335,125 @@ export async function parseCurriculumStructure(filter?: Filter) {
   };
 }
 
-export async function buildCurriculum(lang: string, filters?: Filter) {
-  const contentDir = getContentDir(lang);
+// When in selective mode, we need to ensure that GraphQL is satisfied with basic query data,
+// so we build a minimal full-stack-developer challenge if it wasn't included above.
+async function ensureGraphQLSchema(
+  selectedSuperblockEnums: SuperBlocks[],
+  buildSuperblocks: string | undefined,
+  fullCurriculum: Record<string, unknown>,
+  builder: SuperblockCreator
+) {
+  if (
+    selectedSuperblockEnums.length > 0 &&
+    !buildSuperblocks?.includes('full-stack-developer') &&
+    !fullCurriculum[SuperBlocks.FullStackDeveloper]
+  ) {
+    log(
+      'Building minimal full-stack-developer challenge for GraphQL schema validation.'
+    );
+
+    const { fullSuperblockList: allSuperblocks } =
+      await parseCurriculumStructure();
+    const fsdSuperblock = allSuperblocks.find(
+      sb => sb.name === SuperBlocks.FullStackDeveloper
+    );
+
+    if (fsdSuperblock && fsdSuperblock.blocks.length > 0) {
+      const firstBlock = fsdSuperblock.blocks[0];
+      if (firstBlock) {
+        log(`Building minimal challenge from block: ${firstBlock.dashedName}`);
+        const blockResult = await builder.blockCreator.processBlock(
+          firstBlock,
+          {
+            superBlock: SuperBlocks.FullStackDeveloper,
+            order: 0
+          }
+        );
+        if (blockResult) {
+          fullCurriculum[SuperBlocks.FullStackDeveloper] = {
+            blocks: {
+              [firstBlock.dashedName]: blockResult
+            }
+          };
+        }
+      }
+    }
+  }
+}
+
+async function buildSelectiveCurriculum(
+  lang: string,
+  buildSuperblocks: string,
+  filters?: Filter
+) {
+  const superblockList = buildSuperblocks
+    .split(',')
+    .map(s => s.trim())
+    .filter(s => s.length > 0);
+
+  const availableSuperblocks = Object.keys(superBlockNames).join(', ');
+  const selectedSuperblockEnums: SuperBlocks[] = [];
+
+  for (const superblockName of superblockList) {
+    const enumValue =
+      superBlockNames[superblockName as keyof typeof superBlockNames];
+
+    if (!enumValue) {
+      throw new Error(
+        `Invalid BUILD_SUPERBLOCKS: "${superblockName}". Available superblocks are: ${availableSuperblocks}`
+      );
+    }
+
+    selectedSuperblockEnums.push(enumValue);
+  }
+
+  log(
+    `Selective build mode active: Building full content for ${selectedSuperblockEnums.length} superblock(s): "${superblockList.join('", "')}"`
+  );
+  log(`Other superblocks will be skipped entirely`);
+
+  const firstSuperblock = superblockList[0];
+  const selectiveFilter = { superBlock: firstSuperblock };
```

Comment:
This means that only one superBlock will be built. That's fine for the first iteration, but I would get rid of the code that deals with multiple superBlocks.